src/librustc_lint/redundant_semicolon.rs

```diff
@@ -0,0 +1,40 @@
+use crate::lint::{EarlyLintPass, LintPass, EarlyContext, LintArray, LintContext};
+use syntax::ast::{Stmt, StmtKind, ExprKind};
+use syntax::errors::Applicability;
+
+declare_lint! {
+    pub REDUNDANT_SEMICOLON,
+    Warn,
+    "detects unnecessary trailing semicolons"
+}
+
+declare_lint_pass!(RedundantSemicolon => [REDUNDANT_SEMICOLON]);
+
+impl EarlyLintPass for RedundantSemicolon {
+    fn check_stmt(&mut self, cx: &EarlyContext<'_>, stmt: &Stmt) {
+        if let StmtKind::Semi(expr) = &stmt.node {
+            if let ExprKind::Tup(ref v) = &expr.node {
+                if v.is_empty() {
+                    let multiple = (stmt.span.hi() - stmt.span.lo()).0 > 1;
+                    let msg = if multiple {
+                        "unnecessary trailing semicolons"
+                    } else {
+                        "unnecessary trailing semicolon"
+                    };
+                    let mut err = cx.struct_span_lint(
+                        REDUNDANT_SEMICOLON,
+                        stmt.span,
+                        &msg
+                    );
+                    err.span_suggestion(
+                        stmt.span,
+                        "remove",
```

Comment:
`remove this semicolon` seems better to me.

Reply:
Yeah that's fair, I'll change that